### ISSUE TITLE
nil pointer: index with function

### ISSUE THREAD

**Issue by bodinsamuel** [ariga/atlas#789]
Hi,
I wanted to try the new Partitioning but ended up being blocked by another issue.

I have a table that has an index that uses function:
```sql
CREATE UNIQUE INDEX "test_index" ON "test" ("id", md5("name"));
```

It throw this 
```bash
goroutine 280 [running]:
runtime/debug.Stack()
	/Users/runner/hostedtoolcache/go/1.17.9/x64/src/runtime/debug/stack.go:24 +0x65
runtime/debug.PrintStack()
	/Users/runner/hostedtoolcache/go/1.17.9/x64/src/runtime/debug/stack.go:16 +0x19
github.com/99designs/gqlgen/graphql.DefaultRecover({0xa17f800, 0xa17f700}, {0x4e7a400, 0x6109cb0})
	/Users/runner/go/pkg/mod/github.com/99designs/gqlgen@v0.15.2-0.20220119135911-986650718267/graphql/recovery.go:17 +0x71
github.com/99designs/gqlgen/graphql.(*OperationContext).Recover(0xa35ffff, {0x585aef8, 0xc000390000}, {0x4e7a400, 0x6109cb0})
	/Users/runner/go/pkg/mod/github.com/99designs/gqlgen@v0.15.2-0.20220119135911-986650718267/graphql/context_operation.go:114 +0x3e
ariga.io/atlantis/graphql/exec.(*executionContext)._AtlasSchema_diagram.func1()
	/Users/runner/work/ariga/ariga/atlantis/graphql/exec/exec_gen.go:5585 +0x6a
panic({0x4e7a400, 0x6109cb0})
	/Users/runner/hostedtoolcache/go/1.17.9/x64/src/runtime/panic.go:1038 +0x215
ariga.io/atlantis/graphql/resolver.(*atlasSchemaResolver).Diagram(0xc000126e00, {0x585aef8, 0xc000390000}, 0xc0000f9540)
	/Users/runner/work/ariga/ariga/atlantis/graphql/resolver/atlasSchema.resolvers.go:123 +0xed2
ariga.io/atlantis/graphql/exec.(*executionContext)._AtlasSchema_diagram.func2({0x585aef8, 0xc000390000})
	/Users/runner/work/ariga/ariga/atlantis/graphql/exec/exec_gen.go:5600 +0x82
github.com/99designs/gqlgen/graphql/executor.processExtensions.func4({0x585aef8, 0xc000390000}, 0xc0000f00c0)
	/Users/runner/go/pkg/mod/github.com/99designs/gqlgen@v0.15.2-0.20220119135911-986650718267/graphql/executor/extensions.go:72 +0x26
ariga.io/atlantis/graphql/exec.(*executionContext)._AtlasSchema_diagram(0xc0005879e0, {0x585aef8, 0xc0005aa840}, {0xc0001db280, {0xc000411200, 0x0, 0x0}}, 0xc0000f9540)
	/Users/runner/work/ariga/ariga/atlantis/graphql/exec/exec_gen.go:5598 +0x294
ariga.io/atlantis/graphql/exec.(*executionContext)._AtlasSchema.func8({0x585aef8, 0xc0005aa840})
	/Users/runner/work/ariga/ariga/atlantis/graphql/exec/exec_gen.go:19540 +0xa9
ariga.io/atlantis/graphql/exec.(*executionContext)._AtlasSchema.func9()
	/Users/runner/work/ariga/ariga/atlantis/graphql/exec/exec_gen.go:19545 +0x25
github.com/99designs/gqlgen/graphql.(*FieldSet).Dispatch.func1({0x0, 0xc00012d400})
	/Users/runner/go/pkg/mod/github.com/99designs/gqlgen@v0.15.2-0.20220119135911-986650718267/graphql/fieldset.go:42 +0x3c
created by github.com/99designs/gqlgen/graphql.(*FieldSet).Dispatch
	/Users/runner/go/pkg/mod/github.com/99designs/gqlgen@v0.15.2-0.20220119135911-986650718267/graphql/fieldset.go:41 +0x25d
```

That could probably be expressed like this but it might be complicated to validate the column name:
```hcl

schema "public" {}

table "test" {
  schema = schema.public
  column "id" {
    null = false
    type = int
  }
  column "name" {
    null = true
    type = varchar(100)
  }
  index "index_with_function" {
    on {
      column = column.id
    }
    on {
      column = "md5(name)"
    }
    unique = true
  }
}
```

**Comment by bodinsamuel** [comment]
Forget to say it is happening when inspecting not applying
```atlas schema inspect -d "postgres://<>@localhost:26432/postgres?search_path=public&sslmode=disable" -w```

**Comment by rotemtam** [comment]
Hey @bodinsamuel,

Thanks a lot for the detailed info. A few notes:

1. The HCL block you provided really helps, just a quick note that per the [docs on index expressions](https://atlasgo.io/ddl/sql#index-expressions) to use an index expression use:
```hcl
on {
  expr = "md5(name)"
}
```

2. As we are focused on completely revamping the way the Management UI works in Atlas, we've omitted the UI from the build in [v0.4.0](https://github.com/ariga/atlas/releases/tag/v0.4.0). More news on this to follow soon 

Closing this for now.
